### ISSUE TITLE
Put canvas in different thread from UI

### ISSUE THREAD

**Issue by lazzarello** [evil-mad/stipplegen#8]
This is more of a generalized question about controlP5. I've forked the repo and I'm working on a project which depends on stipplegen, so I'll be giving this some attention but most of that will be learning the processing environment and community.

It seems from the surface that the UI is refreshed at the same frame rate as the canvas. On my target hardware (which is not cutting edge) one iteration with 4000 dots takes about 45 seconds. If I attempt to change the GUI controls for dot size or white point during that 45 seconds, there is no response.

If I hold down the mouse pointer for 45 seconds on the slider, it will "grab" eventually. It would be a nice feature to decouple the controls from the dot rendering.

**Comment by oskay** [comment]
I agree that it would be nice.  My usual method is to pause (if possible) to adjust settings, or to adjust settings before changing the stipple count to be high.

One method would be to only process a certain portion of the voronoi calculation each GUI cycle.

In the long term, I'd prefer to migrate the core graphics library away from toxiclibs, for speed and stability. (No disrespect whatsoever intended to toxiclibs, but there are occasionally memory leaks that come back to haunt us.)
